File: tmp/leetcode75/68.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        def expand(l, r):
            while l >= 0 and r < len(s) and s[l] == s[r]:
                l -= 1; r += 1
                ret[0] += 1
                
        ret = [0]
        for i in range(len(s)):
            expand(i, i)
            expand(i, i+1)
        return ret[0]

File: tmp/leetcode75-c/68.c
static int expand(char *s, int l, int r) {
    int count = 0;
    int slen = strlen(s);
    while (l >= 0 && r < slen && s[l] == s[r]) {
        --l; ++r;
        count += 1;
    }
    return count;
}

int countSubstrings(char * s){
    int slen = strlen(s);
    int ret = 0;
    for (int i = 0; i < slen; ++i) {
        ret += expand(s, i, i);
        ret += expand(s, i, i+1);
    }
    return ret;
}
